src/EDAN35/project/VoxelRenderer.cpp

```diff
@@ -20,65 +20,78 @@ class VoxelRenderer {
         camera = cam;
         this->elapsed_time_s = elapsed_time_s;
 
-        volume = new VoxelVolume(30, 30, 30);
-        // does not work if not rotated like this ??, something wierd in the shader
-        volume->transform.rotateAroundX(glm::pi<float>()*1.5f);
-        volume->transform.scale(3.0f);
-        //volume->transform.translate(Direction::up * 1.0f);
-
-        auto program = GLuint(1u);
+        voxel_program = GLuint(1u);
         shaderManager->CreateAndRegisterProgram(
                 "voxel",
                 {{ShaderType::vertex,   "EDAN35/voxel.vert"},
                  {ShaderType::fragment, "EDAN35/voxel.frag"}},
-                program);
-        volume->setProgram(program);
+                voxel_program);
 
         std::srand(std::time(nullptr));
-        //updateVolume(100.0f);
+        auto tf = Transform()
+                .scale(3.0f)
+                .rotateAroundX(glm::pi<float>() * 1.5f);
+        createVolume(30, tf.translateX(3));
+        createVolume(30, tf.translateX(3));
+        createVolume(30, tf.translateX(3));
+    }
+
+    void createVolume(int size, Transform transform = Transform()) {
+        auto volume = new VoxelVolume(30, 30, 30);
+        // does not work if not rotated like this ??, something wierd in the shader
+        volume->transform = transform;
+        volume->setProgram(voxel_program);
+        volumes.push_back(volume);
+    }
+
+    void update(InputHandler *inputHandler, float dt) {
+        // try this rotation
+        // volume->transform.rotateAroundX(glm::pi<float>()*dt*0.01f);
+        for (auto volume: volumes) {
+            updateVolume(volume);
+        }
     }
 
-    void updateVolume(float elapsed) {
+    void updateVolume(VoxelVolume *volume) {
+        // run whatever algorithm for updating the voxels
+        auto volume_pos = volume->transform.getPos() * glm::vec3(volume->W, volume->H, volume->D);
```

Comment:
When calculating the voxel function we now also use the volume position so that a shape can overlap multiple volumes and not repeat